cmd/rule-evaluator/main.go

```diff
@@ -920,14 +962,72 @@ func newRuleEvaluator(
 	})
 
 	evaluator := ruleEvaluator{
-		rulesManager: rulesManager,
-		queryFunc:    queryFunc,
+		ctx:             ctx,
+		logger:          logger,
+		version:         version,
+		appendable:      appendable,
+		notifierManager: notifierManager,
+		rulesMetrics:    rulesMetrics,
+
+		rulesManager:      rulesManager,
+		queryFunc:         queryFunc,
+		lastEvaluatorOpts: evaluatorOpts,
+		rerunManager:      false,
 	}
 
 	return &evaluator, nil
 }
 
-func (e *ruleEvaluator) ApplyConfig(cfg *config.Config) error {
+func (e *ruleEvaluator) ApplyConfig(cfg *config.Config, evaluatorOpts *evaluatorOptions) error {
+	if evaluatorOpts != nil && !reflect.DeepEqual(evaluatorOpts, e.lastEvaluatorOpts) {
```

Comment:
This is a bit of work that could be replaced by simply detecting change AND os.Exit(0) (with clear log AND metric ideally, AND do this in main ofc).

This has consequence of tricky to debug if the "restart" was actual crash or consequence of valid config changes, but potentially it's fine?